---
bool ok
string errstr
ControllerInfo[] controller_info
